Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2610,6 +2680,113 @@ namespace Babylon
                 }
             }
 
+            Napi::Object GenerateXRInputSourceEvent(Napi::Object& inputSource, Napi::Env env)
+            {
+                auto inputSourceEvent = Napi::Object::New(env);
+                inputSourceEvent.Set("frame", m_jsXRFrame.Value());
+                inputSourceEvent.Set("inputSource", inputSource);
+
+                return inputSourceEvent;
+            }
+
+            void UpdateInputSourceEventValues(
+                std::set<xr::System::Session::Frame::InputSource::Identifier> selectStarts,
+                std::set<xr::System::Session::Frame::InputSource::Identifier> selectEnds,
+                std::set<xr::System::Session::Frame::InputSource::Identifier> squeezeStarts,
+                std::set<xr::System::Session::Frame::InputSource::Identifier> squeezeEnds,
+                Napi::Env env)
+            {
+                for (const auto& id : selectStarts)
+                {
+                    auto inputSourceFound = m_idToInputSource.find(id);
+                    if (inputSourceFound != m_idToInputSource.end())
+                    {
+                        auto inputSourceVal = inputSourceFound->second.Value();
+                        Napi::Object inputSourceEvent = GenerateXRInputSourceEvent(inputSourceVal, env);
+
+                        for (const auto& [name, callback] : m_eventNamesAndCallbacks)
+                        {
+                            if (name == JS_EVENT_NAME_SELECT_START)
+                            {
+                                callback.Call({inputSourceEvent});
+                            }
+                        }
+
+                        m_activeSelects.insert(id);
+                    }
+                }
+
+                for (const auto& id : squeezeStarts)
+                {
+                    auto inputSourceFound = m_idToInputSource.find(id);
+                    if (inputSourceFound != m_idToInputSource.end())
+                    {
+                        auto inputSourceVal = inputSourceFound->second.Value();
+                        Napi::Object inputSourceEvent = GenerateXRInputSourceEvent(inputSourceVal, env);
+
+                        for (const auto& [name, callback] : m_eventNamesAndCallbacks)
+                        {
+                            if (name == JS_EVENT_NAME_SQUEEZE_START)
+                            {
+                                callback.Call({inputSourceEvent});
+                            }
+                        }
+
+                        m_activeSqueezes.insert(id);
+                    }
+                }
+
+                for (const auto& id : selectEnds)
+                {
+                    auto inputSourceFound = m_idToInputSource.find(id);
```

Comment:
>inputSourceFound [](start = 25, length = 16)

nit: consider renaming to like `inputSourceItr` or something to capture that this is an iterator and not a bool.